src/Form/DataTransformer/ModelToIdPropertyTransformer.php

```diff
@@ -110,13 +107,19 @@ public function reverseTransform($value)
 
         foreach ($value as $key => $id) {
             if ('_labels' === $key) {
+                unset($value[$key]);
+
                 continue;
             }
 
-            $collection->add($this->modelManager->find($this->className, $id));
+            $value[$key] = (string) $id;
         }
 
-        return $collection;
+        $query = $this->modelManager->createQuery($this->className);
+        $this->modelManager->addIdentifiersToQuery($this->className, $query, $value);
+        $result = $this->modelManager->executeQuery($query);
+
+        return new ArrayCollection($result);
```

Comment:
Here we are assuming that `executeQuery` is going to [return an `array`](https://github.com/doctrine/collections/blob/fb7a59b4d2c57f7c992428cfd70a0cc501d6f220/lib/Doctrine/Common/Collections/ArrayCollection.php#L58), but that is not the case for [`doctrine/mongodb-odm`](https://github.com/doctrine/mongodb-odm/blob/6bf710befd99ccbf69ec8c7e78dfb2f02613d0a3/lib/Doctrine/ODM/MongoDB/Query/Query.php#L180) where it's usually an `Iterator`.

Reply:
I think we should work to normalize the API as much as possible.
By the moment, I think I could check the type here in order to use the corresponding approach for each case.